src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -374,1042 +376,665 @@
+          receiptFilename =
+            `${this.entityIncNo} - Receipt (Future Effective) - ${this.dateToPacificDate(submittedDate)}.pdf`
         } else {
-          receiptFilename = `${this.entityIncNo} - Receipt - ${filingDate}.pdf`
+          receiptFilename =
+            `${this.entityIncNo} - Receipt - ${this.dateToPacificDate(submittedDate)}.pdf`
         }
 
-        const corpName = this.entityName || this.getCorpTypeNumberedDescription(this.entityType)
-
-        // build filing item
-        const item: HistoryItemIF = {
-          filingType,
-          title: `${this.getCorpTypeDescription(this.entityType)} ${name} - ${corpName}`,
-          filingId: header.filingId,
-          filingAuthor: header.certifiedBy,
-          filingDate,
-          effectiveDate, // used in Future Effective component
-          isFutureEffectiveIa,
-          isFutureEffectiveIaPending,
-          isPaid: (header.status === FilingStatus.PAID),
-          isCompletedIa: (header.status === FilingStatus.COMPLETED),
-          documents: filing?.documents || [] as Array<any>,
-          status: header.status,
-          isCorrected: (header.isCorrected || false),
-          isCorrectionPending: (header.isCorrectionPending || false),
-          comments: this.flattenAndSortComments(header.comments)
+        item.receipt = {
+          type: DocumentTypes.RECEIPT,
+          corpName: this.entityName || this.getCorpTypeNumberedDescription(this.entityType),
+          filingDateTime: filing.submittedDate,
+          paymentToken: filing.paymentToken,
+          title: `Receipt${isFutureEffectiveIa ? ' - Future Effective Incorporation' : ''}`,
+          filename: receiptFilename
         }
+      }
 
-        // add receipt
-        if (header.paymentToken) {
-          item.documents.push({
-            type: this.DOCUMENT_TYPE_RECEIPT,
-            corpName,
-            filingDateTime,
-            paymentToken: header.paymentToken,
-            title: `Receipt${isFutureEffectiveIa ? ' - Future Effective Incorporation' : ''}`,
-            filename: receiptFilename
-          })
-        }
+      this.historyItems.push(item)
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log('Error loading IA filing =', error)
+    }
+  }
 
-        this.historyItems.push(item)
-      } else {
-        // eslint-disable-next-line no-console
-        console.log('ERROR - missing section in filing =', filing)
+  /** Loads a general filing into the historyItems list. */
+  private loadOtherFiling (filing: LedgerIF): void {
+    try {
+      const effectiveDate = new Date(filing.effectiveDate)
+      const submittedDate = new Date(filing.submittedDate)
+
+      // TODO: remove when API already has this in display name
+      let displayName = filing.displayName ||
+        this.filingTypeToName(filing.name) // TODO: delete when API handles unknown filings
```

Comment:
Could this be 1 line or do you prefer a break on an OR?

Reply:
With the comment, it didn't all fit on 1 line.

Anyway, line 634 will go away when the API always returns a displayName (which it currently doesn't), so this expression will then be 1 line.

Actually the whole block should go away when the API gives us the entire displayName.

Reply:
You will find other "TODO"s in the code with a comment such as, "remove when API...".

Sorry, this is currently in an indeterminate state where the API doesn't yet provide all needed functionality, but the current code is usable/testable. I don't expect large changes when the API is complete; most of those TODOs should be easy to resolve.